hpycc/connection.py

```diff
@@ -317,15 +351,64 @@ def run_url_request(self, url, max_attempts, max_sleep):
                 return r
             except (HTTPError, ValueError) as e:
                 attempts += 1
-                time_to_sleep = random.randint(0, max_sleep)
+                time_to_sleep = random.randint(min_sleep, max_sleep)
                 sleep(time_to_sleep)
                 if attempts == max_attempts:
                     raise RetryError(e)
 
+    def get_chunk_from_hpcc(self, logical_file, start_row, n_rows, max_attempts,
```

Comment:
how do this and get-logical_file_chunk work together? do we need both?

Reply:
This part of get-logical_file_chunk is called from more than one place, hence breaking it out. It's used to obtain chunks for data, and also for obtaining the schema. Happy to put it somewhere else but felt nicer than splitting out code. Thoughts?